2019-10-31  Roei Burstein  <classrbu@lnxsrv07.seas.ucla.edu>

	* timespec.h: (timespec_cmp):
	Edited timespec.h to fix error made by saving 8 byte data into 4 byte int
  datatype. Created chain of if statements to compare a.tv_sec and b.tv_sec, and
  if they were equal, compared a.tv_nsec and b.tv_nsec.

--- coreutils-with-bug/lib/timespec.h	2019-10-31 16:37:30.871124000 -0700
+++ coreutils-without-bug/lib/timespec.h	2019-10-31 16:35:58.563627000 -0700
@@ -45,8 +45,29 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec - b.tv_sec < 0)
+    {
+      return -1;
+    }
+  else if (a.tv_sec - b.tv_sec > 0)
+    {
+      return 1;
+    }
+  else
+    {
+      if (a.tv_nsec - b.tv_nsec < 0)
+	{
+	  return -1;
+	}
+      else if (a.tv_nsec - b.tv_nsec > 0)
+	{
+	  return 1;
+	}
+      else
+	{
+	  return 0;
+	}
+    }
 }

 # if ! HAVE_DECL_NANOSLEEP
